components/twitter-card.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import {fallbackLocale} from './locale'
+
+function TwitterCard({
+  card = 'summary_large_image',
+  title = 'Maintainable CSS with React',
+  description = 'React component styling solved with an elegant API, small footprint, and great performance',
+  pathname,
+  locale,
+}) {
+  const urlPrefix = fallbackLocale === locale ? '' : `${locale}.`
+  const image = `https://${urlPrefix}glamorous.rocks/static/images/banner.png`
```

Comment:
Next has a `getInitialProps` method for each page that returns a context with the `req` in it: https://github.com/zeit/next.js/#fetching-data-and-component-lifecycle

I'm not sure about the `getInitialLocaleProps` and if it works on the server. @kentcdodds?

Also, at this point we might want to break this out into a separate issue so we can more easily discuss this than an old PR.